### ISSUE TITLE
Context menu of ElectronClient sidebar is not working for folders

### ISSUE THREAD

**Issue by SamDM** [laurent22/joplin#3873]
## Environment

Joplin version: dev build (49bfa4948966088ee220a4a3ab9552dfc7c36fb5)
Platform: Electron Client Gui
OS specifics: `Linux pop-os 5.4.0-7642-generic #46~1598628707~20.04~040157c-Ubuntu SMP Fri Aug 28 18:02:16 UTC  x86_64 x86_64 x86_64 GNU/Linux`

## Steps to reproduce

1. Compile the development version: `npm install` (49bfa4948966088ee220a4a3ab9552dfc7c36fb5)
2. Run the electron client: `cd ElectronClient && npm start`
3. Make a notebook, right click the notebook in the sidebar.

![image](https://user-images.githubusercontent.com/11174777/95209439-a4967e80-07ea-11eb-9a2e-80196b43f14d.png)

No context menu appears.

## Describe what you expected to happen

The context menu opens.

## Logfile

```
main-html.js:49 Environment: dev
main-html.js:49 appId: net.cozic.joplindev-desktop
main-html.js:49 appType: desktop
main-html.js:45 Electron Security Warning (Insecure Content-Security-Policy) This renderer process has either no Content Security
    Policy set or a policy with "unsafe-eval" enabled. This exposes users of
    this app to unnecessary security risks.

For more information and help, consult
https://electronjs.org/docs/tutorial/security.
This warning will not show up
once the app is packaged.
output.warn @ main-html.js:45
main-html.js:49 CommandService::execute: search Object
SideBar.tsx:242 Uncaught (in promise) TypeError: Cannot read property 'props' of undefined
    at SideBar.tsx:242
    at Generator.next (<anonymous>)
    at fulfilled (/home/sam/Documents/Repos/Personal/joplin/ElectronClient/gui/SideBar/SideBar.js:5)
(anonymous) @ SideBar.tsx:242
fulfilled @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/gui/SideBar/SideBar.js:5
Promise.then (async)
step @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/gui/SideBar/SideBar.js:7
(anonymous) @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/gui/SideBar/SideBar.js:8
__awaiter @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/gui/SideBar/SideBar.js:4
itemContextMenu @ SideBar.tsx:218
callCallback @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:347
invokeGuardedCallbackDev @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:397
invokeGuardedCallback @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:454
invokeGuardedCallbackAndCatchFirstError @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:468
executeDispatch @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:600
executeDispatchesInOrder @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:622
executeDispatchesAndRelease @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:725
executeDispatchesAndReleaseTopLevel @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:733
forEachAccumulated @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:707
runEventsInBatch @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:750
runExtractedPluginEventsInBatch @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:881
handleTopLevel @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:5901
batchedEventUpdates @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:2344
dispatchEventForPluginEventSystem @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:5996
dispatchEvent @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:6025
unstable_runWithPriority @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/scheduler/cjs/scheduler.development.js:643
runWithPriority$2 @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:11305
discreteUpdates$1 @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:21672
discreteUpdates @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:2359
dispatchDiscreteEvent @ /home/sam/Documents/Repos/Personal/joplin/ElectronClient/node_modules/react-dom/cjs/react-dom.development.js:5979
```

**Comment by SamDM** [comment]
An explanation of the problem is: https://www.andrewconnell.com/blog/deal-with-undefined-this-react-event-handler-performant-way/

This is quite easy to fix:

```
diff --git a/ElectronClient/gui/SideBar/SideBar.tsx b/ElectronClient/gui/SideBar/SideBar.tsx
index 20c7c118..5e4119f6 100644
--- a/ElectronClient/gui/SideBar/SideBar.tsx
+++ b/ElectronClient/gui/SideBar/SideBar.tsx
@@ -392,7 +392,7 @@ class SideBarComponent extends React.Component<Props, State> {
                        onFolderDragStart_={this.onFolderDragStart_}
                        onFolderDragOver_={this.onFolderDragOver_}
                        onFolderDrop_={this.onFolderDrop_}
-                       itemContextMenu={this.itemContextMenu}
+                       itemContextMenu={(event:any) => this.itemContextMenu(event)}
                        folderItem_click={this.folderItem_click}
                        onFolderToggleClick_={this.onFolderToggleClick_}
                />;
```

I can create a PR, but I'd like some input first.
1. I noticed that in one place, this fix was already applied: https://github.com/laurent22/joplin/blob/49bfa4948966088ee220a4a3ab9552dfc7c36fb5/ElectronClient/gui/SideBar/SideBar.tsx#L415
2. However, here this is not done: https://github.com/laurent22/joplin/blob/49bfa4948966088ee220a4a3ab9552dfc7c36fb5/ElectronClient/gui/SideBar/SideBar.tsx#L81

Should the proposed fix by applied in case (2.) as well?

Anyway, thanks for this awesome piece of software :smile:

**Comment by laurent22** [comment]
If I'm not mistaken this is already fixed in a recent version.

**Comment by SamDM** [comment]
You're probably right, I think this fixes the issue in the `release 1.2` branch:
https://github.com/laurent22/joplin/blob/559655bf33a13959ceb8edd5ea4a5c7dee7bc7f8/ElectronClient/gui/SideBar/SideBar.tsx#L118
I tried it and it works. I guess this can be closed then.
